boeing_voucher/pdf_parsing.py

```diff
@@ -0,0 +1,153 @@
+"""PDF Parsing functions for Boeing Vouchers"""
+
+import logging
+import pdftotext
+
+log = logging.getLogger()
+
+
+def read_pdf_domestic(pdf):
+    """
+    Process domestic Boeing vouchers and return parsed values
+    """
+    row_values = {"Date": "", "Employee Name": "", "BEMSID": ""}
+
+    column_values = {
+        "Voucher ID": "",
+        "Course Name": "",
+        "Credits/CEUs": "",
+        "Course Dates": "",
+        "Pre-Authorized Amount": "",
+    }
+
+    scanning_rows = False
+    first_row = False
+
+    for page in pdf:
+        for line in page.splitlines():
+            for row_name in row_values:
+                if line.startswith(row_name):
+                    elements = [e.strip() for e in line.split("  ") if e != ""]
+                    if len(elements) > 1:
+                        row_values[row_name] = elements[1]
+
+            if line.startswith("Voucher ID"):
+                start_positions = [line.index(val) for val in column_values]
+                scanning_rows = True
+                first_row = True
+                continue
+            elif line.startswith("NOTE:"):
+                scanning_rows = False
+
+            if scanning_rows:
+                elements = [
+                    line[a:b].strip()
+                    for a, b in zip(start_positions, start_positions[1:] + [None])
+                ]
+                update_column_values(column_values, elements)
+
+                # Handle issue where credits are often incorrectly placed as part of the Course Name column
+                if first_row:
+                    last_val = column_values["Course Name"].split(" ")[-1]
+                    try:
+                        float(last_val)
+                        column_values["Course Name"] = column_values["Course Name"][
+                            0 : column_values["Course Name"].index(last_val)
+                        ]
+                        column_values["Credits/CEUs"] = (
+                            last_val + column_values["Credits/CEUs"]
+                        )
```

Comment:
Seems like the unit tests don't cover the above lines that set the column values, can you add a test pdf that will cause these lines to run?

Reply:
Added a test for this.